src/java.base/windows/native/libnet/NetworkInterface.c

```diff
@@ -213,9 +213,7 @@ int enumInterfaces(JNIEnv *env, netif **netifPP)
                     "IP Helper Library GetIfTable function failed");
                 break;
         }
-        // this different error code is to handle the case when we call
-        // GetIpAddrTable in pure IPv6 environment
-        return -2;
+        return -1;
```

Comment:
*netifPP = NULL;

Reply:
Why?
No users of `enumInterfaces` rely on the value returned in the second parameter when `enumInterfaces` returns a negative result. The same is true for `getAllInterfacesAndAddresses`.
The second parameter was NULL-initialized in a few places; this initialization was not necessary and I just removed it to avoid confusion.

Reply:
ahh... I see your point now. I missed a few cases where the returned value was referenced (free'd) even though the function failed. These should be fixed now.

Reply:
Added null-initialization of return values on function entry.